src/material/button/button.md

```diff
@@ -33,10 +33,46 @@ screen-readers, which will read the text character-by-character. We leave the de
 approach this to the consuming app.
 
 ### Accessibility
-Angular Material uses native `<button>` and `<a>` elements to ensure an accessible experience by
-default. The `<button>` element should be used for any interaction that _performs an action on the
-current page_. The `<a>` element should be used for any interaction that _navigates to another
-view_.
 
+The ARIA spec gives different uses for buttons and links. In order to give the correct information to
+assistive technology, it is important to use the correct one for each situation. 
+
+ - Use the `<button>` element for any interactions that _performs an action on the
+current page_.
+ - Use the `<a>` element for links. That is iteractions that _navigates to another
+view_. This includes navigating to another page, another view in a single-page application and to elements on the same page. Include the href attribute when using the `<a>` element.
+
+#### Links That Look Like Buttons
+Generally, the `<a>` should be used for interactions that perform navigation. It ensures that assistive technology has the
+correct semantic information for the interaction. If a `<button>` must be used, then do the following.
+
+ - Give it the `role='link'` attribute.
+ - Assign an `href` attribute with the link's destination.
```

Comment:
I'm a little confused by this advice. How can a developer assign an `href` attribute to a `<button>`? Even if it were possible, adding `role="link"` + `href` to a `<button>` is not sufficient to truly turn it into a link. For example:

* Links activate only on Enter, buttons activate on Space and Enter.
* Links can open a context menu with right click or the context menu button, buttons cannot.

At best, it may partially surface as a link to SR users, but to non-SR users, it's still very much a button. You might consider just removing the "If a `<button>` must be used .... " section along with the two bullet points, at least until we have a better understanding of _why_ someone would be in a position where they must use  a `<button>` over an `<a>`.